Libraries/pbxbuild/Sources/Tool/SwiftResolver.cpp

```diff
@@ -232,6 +232,73 @@ AppendObjcHeader(
     }
 }
 
+static void
+AppendUnderlyingModule(
+    std::vector<std::string> *args,
+    std::vector<Tool::Invocation::AuxiliaryFile> *auxiliaryFiles,
+    Tool::Context const *toolContext,
+    pbxsetting::Environment const &environment,
+    std::string const &headerName)
+{
+    /* No module map means no underlying module to import. */
+    ext::optional<Tool::ModuleMapInfo::Entry> const &moduleMap = toolContext->moduleMapInfo().moduleMap();
+    if (!moduleMap) {
+        return;
+    }
+
+    /*
+     * Create a module map to import the underlying module. However, exclude the Swift-generated
```

Comment:
The goal is two parts:

 1. Give Swift code in a module access to the Objective-C headers from the same module. This requires a module map for the Objective-C code so Swift can know how to load it.
 2. Automatically load the module's Objective-C half from Swift. This is accomplished with the `-import-underlying-module` flag added below.

The issue is that compiling Swift code also generates an Objective-C header, to let other Objective-C code use the classes, methods, etc from Swift. What this is trying to avoid is circularly loading that header back into Swift as if it was part of the manually written Objective-C module.

For example, let's say you have an Objective-C class `GRPOcean`, and a Swift class `Continent`, both in the same module (i.e. same target). The Swift class should be able to see the Objective-C class without importing it explicitly, and the Objective-C class can see the Swift class by importing the generated Swift header. But we don't want the Swift class to import the generated Swift header, since that defines the same thing the Swift code does!